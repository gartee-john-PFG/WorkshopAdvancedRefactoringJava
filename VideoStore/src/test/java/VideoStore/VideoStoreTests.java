package VideoStore;

import org.junit.jupiter.api.Test;
import org.approvaltests.combinations.CombinationApprovals;

public class VideoStoreTests {
    @Test
    public void approveCustomerStatement() {
        CombinationApprovals.verifyAllCombinations(this::createCustomerStatement
                , new String[] {"Star Wars", "The Maltese Falcon", "101 Dalmations"}
                , new Integer[] {Movie.NEW_RELEASE, Movie.REGULAR, Movie.CHILDRENS}
                , new Integer[] {1, 2, 3, 4});
    }

    private String createCustomerStatement(String movieName, int priceCode, int daysRented) {
        Customer customer = new Customer("Customer Name");
        Movie movie = new Movie(movieName, priceCode);
        movie.setPriceCode(priceCode);
        customer.addRental (new Rental(movie, daysRented));
        return customer.statement();
    }
}