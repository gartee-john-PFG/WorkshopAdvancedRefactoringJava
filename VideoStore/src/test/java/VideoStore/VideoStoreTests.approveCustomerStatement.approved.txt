[Star Wars, 1, 1] => Rental Record for Customer Name
	Star Wars	3.0
Amount owed is 3.0
You earned 1 frequent renter points 
[Star Wars, 1, 2] => Rental Record for Customer Name
	Star Wars	6.0
Amount owed is 6.0
You earned 2 frequent renter points 
[Star Wars, 1, 3] => Rental Record for Customer Name
	Star Wars	9.0
Amount owed is 9.0
You earned 2 frequent renter points 
[Star Wars, 1, 4] => Rental Record for Customer Name
	Star Wars	12.0
Amount owed is 12.0
You earned 2 frequent renter points 
[Star Wars, 0, 1] => Rental Record for Customer Name
	Star Wars	2.0
Amount owed is 2.0
You earned 1 frequent renter points 
[Star Wars, 0, 2] => Rental Record for Customer Name
	Star Wars	2.0
Amount owed is 2.0
You earned 1 frequent renter points 
[Star Wars, 0, 3] => Rental Record for Customer Name
	Star Wars	3.5
Amount owed is 3.5
You earned 1 frequent renter points 
[Star Wars, 0, 4] => Rental Record for Customer Name
	Star Wars	5.0
Amount owed is 5.0
You earned 1 frequent renter points 
[Star Wars, 2, 1] => Rental Record for Customer Name
	Star Wars	1.5
Amount owed is 1.5
You earned 1 frequent renter points 
[Star Wars, 2, 2] => Rental Record for Customer Name
	Star Wars	1.5
Amount owed is 1.5
You earned 1 frequent renter points 
[Star Wars, 2, 3] => Rental Record for Customer Name
	Star Wars	1.5
Amount owed is 1.5
You earned 1 frequent renter points 
[Star Wars, 2, 4] => Rental Record for Customer Name
	Star Wars	3.0
Amount owed is 3.0
You earned 1 frequent renter points 
[The Maltese Falcon, 1, 1] => Rental Record for Customer Name
	The Maltese Falcon	3.0
Amount owed is 3.0
You earned 1 frequent renter points 
[The Maltese Falcon, 1, 2] => Rental Record for Customer Name
	The Maltese Falcon	6.0
Amount owed is 6.0
You earned 2 frequent renter points 
[The Maltese Falcon, 1, 3] => Rental Record for Customer Name
	The Maltese Falcon	9.0
Amount owed is 9.0
You earned 2 frequent renter points 
[The Maltese Falcon, 1, 4] => Rental Record for Customer Name
	The Maltese Falcon	12.0
Amount owed is 12.0
You earned 2 frequent renter points 
[The Maltese Falcon, 0, 1] => Rental Record for Customer Name
	The Maltese Falcon	2.0
Amount owed is 2.0
You earned 1 frequent renter points 
[The Maltese Falcon, 0, 2] => Rental Record for Customer Name
	The Maltese Falcon	2.0
Amount owed is 2.0
You earned 1 frequent renter points 
[The Maltese Falcon, 0, 3] => Rental Record for Customer Name
	The Maltese Falcon	3.5
Amount owed is 3.5
You earned 1 frequent renter points 
[The Maltese Falcon, 0, 4] => Rental Record for Customer Name
	The Maltese Falcon	5.0
Amount owed is 5.0
You earned 1 frequent renter points 
[The Maltese Falcon, 2, 1] => Rental Record for Customer Name
	The Maltese Falcon	1.5
Amount owed is 1.5
You earned 1 frequent renter points 
[The Maltese Falcon, 2, 2] => Rental Record for Customer Name
	The Maltese Falcon	1.5
Amount owed is 1.5
You earned 1 frequent renter points 
[The Maltese Falcon, 2, 3] => Rental Record for Customer Name
	The Maltese Falcon	1.5
Amount owed is 1.5
You earned 1 frequent renter points 
[The Maltese Falcon, 2, 4] => Rental Record for Customer Name
	The Maltese Falcon	3.0
Amount owed is 3.0
You earned 1 frequent renter points 
[101 Dalmations, 1, 1] => Rental Record for Customer Name
	101 Dalmations	3.0
Amount owed is 3.0
You earned 1 frequent renter points 
[101 Dalmations, 1, 2] => Rental Record for Customer Name
	101 Dalmations	6.0
Amount owed is 6.0
You earned 2 frequent renter points 
[101 Dalmations, 1, 3] => Rental Record for Customer Name
	101 Dalmations	9.0
Amount owed is 9.0
You earned 2 frequent renter points 
[101 Dalmations, 1, 4] => Rental Record for Customer Name
	101 Dalmations	12.0
Amount owed is 12.0
You earned 2 frequent renter points 
[101 Dalmations, 0, 1] => Rental Record for Customer Name
	101 Dalmations	2.0
Amount owed is 2.0
You earned 1 frequent renter points 
[101 Dalmations, 0, 2] => Rental Record for Customer Name
	101 Dalmations	2.0
Amount owed is 2.0
You earned 1 frequent renter points 
[101 Dalmations, 0, 3] => Rental Record for Customer Name
	101 Dalmations	3.5
Amount owed is 3.5
You earned 1 frequent renter points 
[101 Dalmations, 0, 4] => Rental Record for Customer Name
	101 Dalmations	5.0
Amount owed is 5.0
You earned 1 frequent renter points 
[101 Dalmations, 2, 1] => Rental Record for Customer Name
	101 Dalmations	1.5
Amount owed is 1.5
You earned 1 frequent renter points 
[101 Dalmations, 2, 2] => Rental Record for Customer Name
	101 Dalmations	1.5
Amount owed is 1.5
You earned 1 frequent renter points 
[101 Dalmations, 2, 3] => Rental Record for Customer Name
	101 Dalmations	1.5
Amount owed is 1.5
You earned 1 frequent renter points 
[101 Dalmations, 2, 4] => Rental Record for Customer Name
	101 Dalmations	3.0
Amount owed is 3.0
You earned 1 frequent renter points 
